src/main/java/com/dabsquared/gitlabjenkins/util/CommitStatusUpdater.java

```diff
@@ -64,7 +72,15 @@ private static void printf(TaskListener listener, String message, Object... args
     }
 
     private static String getBuildRevision(Run<?, ?> build) {
-        return build.getAction(BuildData.class).getLastBuiltRevision().getSha1String();
+        BuildData action = build.getAction(BuildData.class);
+        try {
+            Field lastBuildField = action.getClass().getDeclaredField("lastBuild");
+            lastBuildField.setAccessible(true);
+            Build lastBuild = (Build) lastBuildField.get(action);
+            return lastBuild.getMarked().getSha1String();
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            return null;
+        }
     }
```

Comment:
Wow, i am blind. Or i really got familiar with fields are never public in
Java. :)

On 19 May 2016 at 16:16, Martin Nowak <notifications@github.com> wrote:

> In
> src/main/java/com/dabsquared/gitlabjenkins/util/CommitStatusUpdater.java
> <https://github.com/jenkinsci/gitlab-plugin/pull/311#discussion_r63886055>
> :
>
> > @@ -64,7 +72,15 @@ private static void printf(TaskListener listener, String message, Object... args
> >      }
> >
> >      private static String getBuildRevision(Run<?, ?> build) {
> > -        return build.getAction(BuildData.class).getLastBuiltRevision().getSha1String();
> > +        BuildData action = build.getAction(BuildData.class);
> > +        try {
> > +            Field lastBuildField = action.getClass().getDeclaredField("lastBuild");
> > +            lastBuildField.setAccessible(true);
> > +            Build lastBuild = (Build) lastBuildField.get(action);
> > +            return lastBuild.getMarked().getSha1String();
> > +        } catch (NoSuchFieldException | IllegalAccessException e) {
> > +            return null;
> > +        }
> >      }
>
> lastBuild is a public field of BuildData, and up to now there was no null
> checking for the result of getLastBuiltRevision, so it seems to be always
> setl.
>
> https://github.com/jenkinsci/git-plugin/blob/7ea2ad06da30aa6778b33fdd4c1f94389c20fbee/src/main/java/hudson/plugins/git/util/BuildData.java#L41
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/gitlab-plugin/pull/311/files/772b9f9c08ba7fdc3a73fa025b45f4213b0dd01b#r63886055>
>